### PR TITLE
added transport to options so it can be used to filter by

### PR DESCRIPTION
**Pull request by SidHarder**

### CODE REVIEW

**Comment by SidHarder** [comment]
question: Was transport left out on purpose for some reason?

**Comment by diasdavid** [comment]
The toOptions is a convenience method so that we get an options object that is friendly with Node.js net and http api. we can add this one in.

Seems like tests are failing, could you update?

**Comment by SidHarder** [comment]
I take a look and see whats up with the tests.

**Comment by SidHarder** [comment]
Tests are fixed.

File: test/index.spec.js [hunk 1]
```diff
@@ -302,6 +302,7 @@ describe('helpers', () => {
       ).to.be.eql({
         family: 'ipv4',
         host: '0.0.0.0',
+				transport: 'tcp',
```
Comment (diasdavid) [inline test/index.spec.js]: Fails linting:
https://travis-ci.org/multiformats/js-multiaddr/jobs/176772945#L1162-L1163

**Comment by diasdavid** [comment]
@SidHarder you might also want to rebase master onto your branch, so that you don't get the weird Webpack errors (the loader ones)

File: test/index.spec.js (hunk 1 above)
Comment (SidHarder) [inline test/index.spec.js]: I rebased master onto my branch but still have same issue?

Comment (SidHarder) [inline test/index.spec.js]: OK, I got through it, disregard the previous messages.

**Comment by diasdavid** [comment]
released :)

**Pull request merged**